src/injector/inject.ts

```diff
@@ -358,16 +380,39 @@ export function inject (vm: EurekaCompatibleVM) {
     }
 
     // Blockly stuffs
-    vm.once('workspaceUpdate', () => {
-        const blockly = (window.eureka.blockly = getBlocklyInstance(vm));
-        if (!blockly) {
+    let initalized = false;
+    getBlocklyInstance(vm).then((blockly) => {
+        if (!initalized) {
```

Comment:
How do you make sure you actually get blockly instance here?

Reply:
If `EXTENSION_ADDED` event does not exist or has no `ScratchBlocks`, the promise will not resolve.